lib/restbase.js

```diff
@@ -49,10 +52,40 @@ function RESTBase (options, req) {
             router: options.router,
             statsd: options.metrics
         };
-
     }
+
+    // set the request id, if appropriate
+    this.setRequestId(req);
+
 }
 
+// sets the request id for this instance and adds it to
+// the request header, if defined
+RESTBase.prototype.setRequestId = function(req) {
+
+    if (this._isMain) {
```

Comment:
Right, but the problem is that when we create the main object, `req` is not set, and that child instances can have one single request ID, the parent cannot, precisely because it's used for all requests. Hence, if this is the main RB object, and we have a `req`, set only the header, otherwise set the reqid of the RB object (which is a child of the main obj), so that any sub-requests may inherit the value regardless of which request is passed to them.